homeassistant/core.py

```diff
@@ -1184,7 +1186,14 @@ def async_set(
         This method must be run in the event loop.
         """
         entity_id = entity_id.lower()
-        new_state = str(new_state)
+        if isinstance(cast(Any, new_state), float):
+            # If the entity's state is a float, limit precision according to machine
+            # epsilon to make the string representation readable
+            epsilon = sys.float_info.epsilon
+            useful_precision = abs(int(math.floor(math.log10(abs(epsilon)))))
```

Comment:
This only needs to be calculated once, if the concept is viable this should be stowed away somewhere.